lib/volcanic/authenticator/v1/warden/auth_failure.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Volcanic::Authenticator
+  module V1::Warden
```

Comment:
probably better to leave this to each service to implement. The auth gem can easily provide information to say "this is authenticated or not", but it doesn't have responsibility for client interactions.

Auth gem doesn't know things like:
* what language the user needs an error response in
* what the site's error page should be
* what content type the client should be served (are they a browser? are they an API caller? did they request XML?)

Reply:
OK then we can have this class customisable. meaning we can pass status, header or body that we want

Reply:
Just that if the service want to use the default failure setting, it can just call the class with minimum setup. 
```ruby
# use default setup
confg.failure_app = AuthFailure.new

# to customise
config.failure_app = AuthFailure.new(status: 403, headers: { 'Content-Type' => 'application/xml' }, body: {error: 'custom message'})
```